### ISSUE TITLE
🐛 Bug Report: Navigation via VoiceOver screenreader does not announce page change to user

### ISSUE THREAD

**Issue by matthewpennell** [backstage/backstage#17456]
### 📜 Description

When navigating between plugins while using the VoiceOver screenreader software, selecting an item from the navigation panel loads the plugin, but no announcement or change of focus happens, meaning that a non-sighted screenreader user will be unaware what has happened or how to proceed.

### 👍 Expected behavior

Whenever a new page loads, it should be announced to screenreader assistive technology. Either by moving the current focused element to an element that describes the new page (e.g. the header or first heading within the plugin content), or by triggering a status message targeted at screenreader users.

### 👎 Actual Behavior with Screenshots

No screenshots, but I can make a screen recording if necessary. Or just test it yourself on a Mac.

### 👟 Reproduction steps

- On a Mac, start VoiceOver by pressing Cmd+F5
- Open a page on Backstage and give it focus by clicking somewhere in the page
- Navigate sequentially through the page content with Ctrl+Opt+left arrow or right arrow
- On any navigation item, activate the link with Ctrl+Opt+Space
- Notice that the new page loads but nothing is announced by VoiceOver

(Press Cmd+F5 to turn VoiceOver off again)

### 📃 Provide the context for the Bug.

Any users of VoiceOver (and probably all other screenreaders, but I don't have them available to test with) will be severely inconvenienced by this issue. Non-sighted users may struggle to understand what is happening on the page, or even what page they are currently on.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else
